src/Processors/Transforms/FillingTransform.cpp

```diff
@@ -100,9 +109,12 @@ static bool tryConvertFields(FillColumnDescription & descr, const DataTypePtr &
         || descr.fill_step.getType() > max_type)
         return false;
 
-    descr.fill_from = convertFieldToType(descr.fill_from, *to_type);
-    descr.fill_to = convertFieldToType(descr.fill_to, *to_type);
-    descr.fill_step = convertFieldToType(descr.fill_step, *to_type);
+    if (!descr.fill_from.isNull())
```

Comment:
This is much better - we are explicitly asking if the field is convertible, instead of having a risk it is confused with NULL, which has another meaning for the WITH FILL logic.